fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisJob.java

```diff
@@ -0,0 +1,171 @@
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.StringJoiner;
+
+public class AnalysisJob {
+
+    public static final Logger LOG = LogManager.getLogger(AnalysisJob.class);
+
+    protected Set<BaseAnalysisTask> queryingTask;
+
+    protected Set<BaseAnalysisTask> queryFinished;
+
+    protected List<ColStatsData> buf;
+
+    protected int total;
+
+    protected int acc;
+
+    protected StmtExecutor stmtExecutor;
+
+    protected boolean killed;
+
+    protected long start;
+
+    protected AnalysisInfo jobInfo;
+
+    public AnalysisJob(AnalysisInfo jobInfo, Collection<? extends BaseAnalysisTask> queryingTask) {
+        for (BaseAnalysisTask task : queryingTask) {
+            task.job = this;
+        }
+        this.queryingTask = new HashSet<>(queryingTask);
+        this.queryFinished = new HashSet<>();
+        this.buf = new ArrayList<>();
+        total = queryingTask.size();
+        start = System.currentTimeMillis();
+        this.jobInfo = jobInfo;
+    }
+
+    public synchronized void appendBuf(BaseAnalysisTask task, List<ColStatsData> statsData) {
+        queryingTask.remove(task);
+        buf.addAll(statsData);
+        queryFinished.add(task);
+        acc += 1;
+        if (acc == total) {
+            writeBuf();
+            updateTaskState(AnalysisState.FINISHED, "Cost time in sec: "
+                    + (System.currentTimeMillis() - start) / 1000);
+            deregisterJob();
+        } else if (buf.size() >= StatisticConstants.ANALYZE_JOB_BUF_SIZE) {
+            writeBuf();
+        }
+    }
+
+    // CHECKSTYLE OFF
+    // fallthrough here is expected
+    public void updateTaskState(AnalysisState state, String msg) {
+        long time = System.currentTimeMillis();
+        switch (state) {
+            case FAILED:
+                for (BaseAnalysisTask task : queryingTask) {
+                    Env.getCurrentEnv().getAnalysisManager().updateTaskStatus(task.info, state, msg, time);
+                }
+            case FINISHED:
+                for (BaseAnalysisTask task : queryFinished) {
+                    Env.getCurrentEnv().getAnalysisManager().updateTaskStatus(task.info, state, msg, time);
+                }
+            default:
+                // DO NOTHING
+        }
+    }
+
+    protected void writeBuf() {
+        String insertStmt = "INSERT INTO " + StatisticConstants.FULL_QUALIFIED_STATS_TBL_NAME + " VALUES ";
+        StringJoiner values = new StringJoiner(",");
+        for (ColStatsData data : buf) {
+            values.add(data.toSQL(true));
+        }
+        insertStmt += values.toString();
+        try (AutoCloseConnectContext r = StatisticsUtil.buildConnectContext(false)) {
+            stmtExecutor = new StmtExecutor(r.connectContext, insertStmt);
+            executeWithExceptionOnFail(stmtExecutor);
```

Comment:
How about add retry logic here? Because multiple tasks will fail if this insert fail to finish.